server/routes/api/checkout.js

```diff
@@ -41,27 +41,54 @@ router.post('/create-transaction', async (req, res) => {
 // 3. Once user completes payment, will redirect back to `success_url` with
 //  a Stripe session_id included in the URL.
 
-router.post('/create-checkout-session', async (req, res) => {
-  try {
-    const acceptableCharges = [1, 2, 20, 50]
-    const { donationAmount } = req.body || {}
-    const parsedDonationAmount = parseInt(donationAmount, 10)
+router.post('/create-checkout-session', async (req, res) => {  
+  const { donationAmount } = req.body || {}
+  const origin = req.get('origin')
 
-    let donation
+  const input = format(donationAmount);
+  const [inputIsValid, message] = validate(input);
+  const donationAmountForStripe = input * 100; // Stripe accepts values in cents
 
-    if (parsedDonationAmount < 2) {
-      // TODO: Change to something better later.
-      donation = 150
-    } else {
-      donation = parsedDonationAmount * 100
-    }
+  if (inputIsValid) {
+    const session = checkout(donationAmountForStripe, origin);
+    // console.log('session:', session)
 
-    if (!acceptableCharges.includes(parsedDonationAmount)) {
-      return res.status(400).send({ error: 'Invalid Amount' })
+    if (session.error) {
+      console.log({ error: session.error, message: 'An error occurred with Stripe checkout' });
+      return res.json(session);
     }
 
-    const origin = req.get('origin')
+    return res.json({ url: session.url, sessionId: session.id })
+  } else {
+    return res.status(400).send({ error: 'Invalid Amount; did not create Stripe checkout session', message })
+  }
+})
+
+// format input value
+function format(value) {
```

Comment:
Good idea to abstract these validation and formatting methods from the route. Since you're also using them in the Vue component, you could create a directory for utility code and import the methods from there. That will dry this up a little bit.